Sources/Sentry/SentryExtraContextProvider.m

```diff
@@ -44,6 +44,23 @@ - (NSDictionary *)getExtraDeviceContext
     extraDeviceContext[SentryDeviceContextFreeMemoryKey] = @(self.crashWrapper.freeMemorySize);
     extraDeviceContext[@"processor_count"] = @([self.processInfoWrapper processorCount]);
 
+    switch([self.processInfoWrapper thermalState]){
+    case NSProcessInfoThermalStateNominal:
+        extraDeviceContext[@"thermal_state"] = @"nominal";
+        break;
+    case NSProcessInfoThermalStateFair:
+        extraDeviceContext[@"thermal_state"] = @"fair";
+        break;
+    case NSProcessInfoThermalStateSerious:
+        extraDeviceContext[@"thermal_state"] = @"serious";
+        break;
+    case NSProcessInfoThermalStateCritical:
+        extraDeviceContext[@"thermal_state"] = @"critical";
+        break;
+    default:
+        break;
+    }
```

Comment:
h: The only thing that I would change is to extract this naming from here.
Something like what we have in [here](https://github.com/getsentry/sentry-cocoa/blob/main/Sources/Sentry/SentryDataCategoryMapper.m).

Reply:
And to align with future android plans we need to change the names:

NSProcessInfoThermalStateNominal = "light"
NSProcessInfoThermalStateFair = "moderate"
NSProcessInfoThermalStateSerious = "severe"
NSProcessInfoThermalStateCritical = "critical
"

Reply:
reading the [iOS](https://developer.apple.com/documentation/foundation/nsprocessinfothermalstate) and [Android](https://developer.android.com/reference/android/os/PowerManager#THERMAL_STATUS_NONE) docs, I think these don't correlate. on iOS it's more about the thermal state, whereas on Android it's about how severely system is affected by thermal state. e.g. `moderate` on Android means "Thermal status code: Moderate throttling where UX is not largely impacted." which means it's already elevated and impacts UX at least to some extent.

TLDR: I wouldn't try to match/correlate and instead just report as-is

Reply:
Ok. Lets keep the name as is then.

Reply:
@brustolin updated to extract naming to constants at the top. Do you also want a separate file that does the mapping or are the extracted strings enough? I based it on the phone orientation property right below it

Reply:
All good, thanks for the help.